pallets/funding/src/functions.rs

```diff
@@ -3164,6 +3179,71 @@ impl<T: Config> Pallet<T> {
 		Ok((reward_info, evaluations_count))
 	}
 
+	pub fn generate_long_term_holder_reward_amount(project_id: ProjectId) -> Result<BalanceOf<T>, DispatchError> {
+		// Fetching the necessary data for a specific project.
+		let project_details = ProjectsDetails::<T>::get(project_id).ok_or(Error::<T>::ProjectNotFound)?;
+		let project_metadata = ProjectsMetadata::<T>::get(project_id).ok_or(Error::<T>::ProjectNotFound)?;
+
+		// Determine how much funding has been achieved.
+		let funding_amount_reached = project_details.funding_amount_reached;
+		let fundraising_target = project_details.fundraising_target;
+		let total_issuer_fees = Self::calculate_fees(funding_amount_reached);
+
+		let initial_token_allocation_size = project_metadata.total_allocation_size;
+		let final_remaining_contribution_tokens = project_details.remaining_contribution_tokens;
```

Comment:
The preamble of these two new functions (`generate_long_term_holder_reward_amount` and `generate_liquidity_pools_reward_amount`) looks like identical, and they are called one after the other. Do we actually have to compute these values fresh every time, or can we cache the compution made by the first function and pass it to the second?

Reply:
Yea I chose readability over DRY on this one. If you don't like it I can have one function returning both values or like you said have the values precomputed and passed to the functions

Reply:
Ok, but what about the `total_issuer_fees`? Isn't that computed two times by executing `calculate_fees`?

Reply:
Alright, I'm going to combine both functions into one, since we call them always together